src/main/java/com/amazon/opendistroforelasticsearch/security/ssl/transport/OpenDistroSSLDualModeConfig.java

```diff
@@ -0,0 +1,59 @@
+/*
+ * Copyright 2020 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License").
+ * You may not use this file except in compliance with the License.
+ * A copy of the License is located at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * or in the "license" file accompanying this file. This file is distributed
+ * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
+package com.amazon.opendistroforelasticsearch.security.ssl.transport;
+
+import com.amazon.opendistroforelasticsearch.security.support.ConfigConstants;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import org.elasticsearch.common.settings.ClusterSettings;
+import org.elasticsearch.common.settings.Setting;
+import org.elasticsearch.common.settings.Settings;
+
+public class OpenDistroSSLDualModeConfig {
+
+    public static final Setting<Boolean> SSL_DUAL_MODE_SETTING = Setting.boolSetting(ConfigConstants.OPENDISTRO_SECURITY_SSL_DUAL_MODE_ENABLED,
+            true, Setting.Property.NodeScope, Setting.Property.Dynamic, Setting.Property.Filtered);
+
+    private static final Logger logger = LogManager.getLogger(OpenDistroSSLDualModeConfig.class);
+
+    private final boolean isSSLOnly;
+    private volatile boolean dualModeEnabled;
+
+    public OpenDistroSSLDualModeConfig(final Settings settings) {
+        isSSLOnly = settings.getAsBoolean(ConfigConstants.OPENDISTRO_SECURITY_SSL_ONLY, false);
+        dualModeEnabled = settings.getAsBoolean(ConfigConstants.OPENDISTRO_SECURITY_SSL_DUAL_MODE_ENABLED,
+                false);
+        logger.info("SSL dual mode is {}", isDualModeEnabled() ? "enabled" : "disabled");
+    }
+
+    public void registerClusterSettingsChangeListener(final ClusterSettings clusterSettings) {
+        clusterSettings.addSettingsUpdateConsumer(SSL_DUAL_MODE_SETTING,
```

Comment:
If it is registered as a `ClusterSettings` is it necessary to have a separate API to manage it? Why this can't be done using ES cluster settings API?

Reply:
Idea is to ensure that we provide abstraction over the settings i.e do additional validation/business logic etc. In future we need to add additional validation etc i.e admin user can only set these API's etc for FGAC domain's etc, this can allow us to accomplish it.

Reply:
It is possible to change the settings using ES cluster API and having two different ways to change the settings is confusing, IMO. If it can be done using standard ES cluster settings API, there is no need to introduce (and maintain) another API.

Reply:
Yes, for current ssl_only mode it can be changed, but idea is to ensure that dynamic enabling/disabling can be allowed only by privileged users. Currently for only_ssl mode there isn't much of a benefit like you mentioned, but if we are enabling this in FGAC mode (there would be additional configuration change to it), it makes sense to control who can dynamically change the property as well as we can change backend settings store used (using security index), i.e similar to how we made nodes_dn dynamic. Let us know ur opinion.

Reply:
Let's introduce new API when necessary, for now I would prefer standard ES cluster settings API.

Reply:
Removed the PUT apis, we can use the cluster settings api to set the dual mode value. Did not remove the GET api since the cluster settings are filtered out by default.

Reply:
Removed the GET api also and updated the ```opendistro_security.ssl_dual_mode_enabled``` to not be a Filtered setting. 
Also, removed ```getSettingsFilter()``` function override in ```OpenDistroSecurityPlugin``` and ```OpenDistroSecuritySSLPlugin``` classes, which was filtering out all ```opendistro_security.*``` settings. 
Instead added explicit ```Property.Filtered``` to every other setting which didn't have it.